### PR TITLE
Refactor/uncommited txs

### LINKED ISSUE #135: Refactor useUncommitedTxs

### PR DESCRIPTION
**Pull request by JordiParraCrespo**
# Refactor/uncommited txs
## Issues
- closes #135  
## PR deps
- #131 
## Changes
- Refactor uncommited txs to improve reusability

### CODE REVIEW

**Review by AgustinMJ — CHANGES_REQUESTED** [review]

File: src/module/transaction/hook/useAddUncommitedTransaction.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { WalletStorage } from "module/wallet/WalletStorage";
+import { NetworkType } from "module/settings/state/SettingsState";
+import useWalletState from "module/wallet/hook/useWalletState";
+import useUpdateUncommittedTransactions from "./useUpdateUncommitedTransactions";
+
+const useAddUncommittedTransaction = (): ((index: number, chain: NetworkType, hash: string) => Promise<void>) => {
+    const {
+        state: { wallets },
+    } = useWalletState();
+    const updateUnCommitedTxsHashes = useUpdateUncommittedTransactions();
```
[the remark this answers is not in the document]
Comment (JordiParraCrespo) [reply src/module/transaction/hook/useAddUncommitedTransaction.ts:10]: fixed

File: src/module/transaction/hook/useUpdateUncommitedTransactions.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import useServiceInstance from "module/wallet/hook/useServiceInstance";
+import useWalletState from "module/wallet/hook/useWalletState";
+import updateWalletUncommittedTxHashes from "module/wallet/utils/updateWalletUncommittedTxHashes";
+
+export default function useUpdateUncommittedTransactions() {
```
[the remark this answers is not in the document]
Comment (JordiParraCrespo) [reply src/module/transaction/hook/useUpdateUncommitedTransactions.ts:5]: done

File: src/module/transaction/query/useUncommittedTransactions.ts [hunk 3]
```diff
@@ -1,22 +1,36 @@
...
-    const setWalletState = useSetRecoilState(walletState);
+    /**
+     * We need to use a ref here because the query will be re-run
+     * and we don't want to remove the hashes from the storage on each re-run
+     * but we want to display the rejected txs that we have uncommited in that session
+     */
     const rejectedHashes: string[] = useRef([]).current;
-
+    const updateUncommitedTxsHashes = useUpdateUncommittedTransactions();
+    const removeUTxFromStorage = (hash: string): Promise<void> => {
```
[the remark this answers is not in the document]
Comment (JordiParraCrespo) [reply src/module/transaction/query/useUncommittedTransactions.ts:21]: done

File: src/module/wallet/WalletStorage.ts [hunk 4]
```diff
@@ -104,46 +105,26 @@ export const WalletStorage = new (class extends BaseStorageService<SecureWalletS
...
 
+    async addUncommittedTransactionHash(index: number, chain: Chain, hash: string): Promise<void> {
+        const storage = await this.get();
+        if (!storage) return;
+        const uncommittedTransactionHashes = (await this.getUncommittedTransactionHashes(index, chain)) || [];
+        return this.updateUncommitedTransactionHashes(index, chain, [...uncommittedTransactionHashes, hash]);
+    }
+
     async removeUncommittedTransactionHash(index: number, chain: Chain, hash: string): Promise<void> {
         const storage = await this.get();
```
[the remark this answers is not in the document]
Comment (JordiParraCrespo) [reply src/module/wallet/WalletStorage.ts:123]: talked

**Review by AgustinMJ — APPROVED** [review]

**Pull request merged**